packager/packager.h

```diff
@@ -83,7 +83,8 @@ class Packager final {
                                             std::vector<ast::ParsedFile> files);
 
     // Run packager incrementally. Note: `files` must contain all packages files. Does not support package changes.
-    static std::vector<ast::ParsedFile> runIncremental(core::GlobalState &gs, std::vector<ast::ParsedFile> files);
+    template <typename StateType>
+    static std::vector<ast::ParsedFile> runIncremental(StateType &gs, std::vector<ast::ParsedFile> files);
```

Comment:
Slight preference for being explicit here with two different declarations of `runIncremental` and leaving the template magic in the `.cc` file: it's easier to catch mistakes in client code that way (e.g. the current state would happily accept you passing in a `unique_ptr<GlobalState>`...or a `MutableCtx`, which would normally auto-convert to `GlobalState`, but not here.  you wouldn't see an error until link time).

(The proposed scheme here is also inconsistent with namer and resolver: the non-mutating versions of those have different names, whereas packager would not.  I tend to think having identical names is not a good thing for what we're trying to do, even if C++ is happy to overload on `const` qualification.)